libnmstate/nm/bridge_port_vlan.py

```diff
@@ -0,0 +1,251 @@
+            LB.Port.Vlan.TRUNK_TAGS: self.trunk_ports,
+        }
+        if self.tag:
+            port_vlan_state[LB.Port.Vlan.TAG] = self.tag
+        if self.port_type == LB.Port.Vlan.Mode.TRUNK:
+            port_vlan_state[
+                LB.Port.Vlan.ENABLE_NATIVE
+            ] = self.enable_native_vlan
+        return port_vlan_state
+
+    def import_from_bridge_settings(self, port_vlans):
+        """
+        Instantiates a PortVlanFilter object from a list of NM.BridgeVlan
+        objects.
+        """
+
+        bridge_vlan_config = [
+            PortVlanFilter._get_vlan_info(port_vlan_filter_entry)
+            for port_vlan_filter_entry in port_vlans
+        ]
+
+        trunk_tags, tag, enable_native = PortVlanFilter._get_vlan_config(
+            bridge_vlan_config
+        )
+
+        self._trunk_ports = trunk_tags
+        self._tag = tag
+        self._is_native_vlan = enable_native
+        self._port_type = (
+            LB.Port.Vlan.Mode.TRUNK if trunk_tags else LB.Port.Vlan.Mode.ACCESS
+        )
+
+    @staticmethod
+    def get_vlan_tag_range(port_vlan):
+        """
+        Extract the vlan tags from the NM.BridgeVlan object.
+        A single NM.BridgeVlan object can have a range of vlan tags, or a
+        single one.
+        When a NM.BridgeVlan holds a single tag, the min_range, and max_range
+        returned will have the same vlan tag.
+        :return: min_range, max_range
+        """
+
+        port_vlan_tags = port_vlan.to_str().split()
+
+        if "-" in port_vlan_tags[0]:
+            vlan_min, vlan_max = port_vlan_tags[0].split("-")
+            return int(vlan_min), int(vlan_max)
+        else:
+            tag = port_vlan_tags[0]
+            return int(tag), int(tag)
+
+    @staticmethod
+    def _is_access_port(trunk_tags):
+        if (
+            len(trunk_tags) == 1
+            and trunk_tags[0][PortVlanFilter.PVID]
+            and trunk_tags[0][PortVlanFilter.UNTAGGED]
+        ):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def _get_vlan_info(port_vlan_info):
+        vlan_min, vlan_max = PortVlanFilter.get_vlan_tag_range(port_vlan_info)
+        pvid = port_vlan_info.is_pvid()
+        untagged = port_vlan_info.is_untagged()
+        if vlan_max != vlan_min:
+            port_data = {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: vlan_min,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: vlan_max,
+                }
+            }
+        else:
+            port_data = {LB.Port.Vlan.TrunkTags.ID: vlan_min}
+        port_data.update(pvid=pvid, untagged=untagged)
+        return port_data
+
+    @staticmethod
+    def _generate_vlan_trunk_port_config(trunk_port):
+        single_vlan_tag = trunk_port.get(LB.Port.Vlan.TrunkTags.ID)
+        ranged_vlan_tags = trunk_port.get(LB.Port.Vlan.TrunkTags.ID_RANGE)
+        port_vlan = None
+
+        if single_vlan_tag:
```

Comment:
Not sure if you can assume that one of these keys do exist in the state, but assuming you do:
```
min_range = max_range = trunk_port.get(LB.Port.Vlan.TrunkTags.ID)
if min_range is None:
    range = trunk_port.get(LB.Port.Vlan.TrunkTags.ID_RANGE)
    min_range = ranged_vlan_tags[LB.Port.Vlan.TrunkTags.MIN_RANGE]
    max_range = ranged_vlan_tags[LB.Port.Vlan.TrunkTags.MAX_RANGE]
port_vlan = nmclient.NM.BridgeVlan.new(min_range, max_range)
...
```